lessons/ro/chapter_6.yaml

```diff
@@ -0,0 +1,188 @@
+    Un subșir de caractere (în engleză *string slice*) este o referință la o secvență de octeți din memorie ce trebuie
+    întotdeauna să fie în format utf-8 valid.
+
+    Un subșir al unui subșir (în engleză *sub-slice*) de `str`, trebuie să fie, de asemenea, în format utf-8 valid.
+
+
+    Metode comune ale `&str`:
+
+    * `len` obține lungimea șirului literal în octeți (nu numărul de caractere).
+
+    * `starts_with`/`ends_with` pentru teste de bază.
+
+    * `is_empty` returnează *true* dacă lungimea este zero.
+
+    * `find` returnează un `Option<usize>` al primei poziții dintr-un text.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20main()%20%7B%0A%20%20%20%20let%20a%20%3D%20%22salut%20%F0%9F%A6%80%22%3B%0A%20%20%20%20println!(%22%7B%7D%22%2C%20a.len())%3B%0A%20%20%20%20let%20primul_cuvant%20%3D%20%26a%5B0..5%5D%3B%0A%20%20%20%20let%20al_doilea_cuvant%20%3D%20%26a%5B6..10%5D%3B%0A%20%20%20%20%2F%2F%20let%20jumatate_de_crab%20%3D%20%26a%5B6..8%5D%3B%20E%C8%98UEAZ%C4%82%0A%20%20%20%20%2F%2F%20Rust%20nu%20accepta%20subsiruri%20formate%20din%20caractere%20Unicode%20invalide%0A%20%20%20%20println!(%22%7B%7D%20%7B%7D%22%2C%20primul_cuvant%2C%20al_doilea_cuvant)%3B%0A%7D%0A
+- title: Tipul Char
+  content_markdown: >
+    Cu atâtea dificultăți în lucrul cu Unicode, Rust oferă o modalitate de a
+    prelua o secvență de octeți utf-8 ca un vector de caractere de tip `char`.
+
+
+    Un `char` are întotdeauna o lungime de 4 octeți (ceea ce permite o căutare eficientă după caractere individuale).
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20main()%20%7B%0A%20%20%20%20%2F%2F%20colecteaza%20caracterele%20ca%20vector%20de%20char-uri%0A%20%20%20%20let%20chars%20%3D%20%22salut%20%F0%9F%A6%80%22.chars().collect%3A%3A%3CVec%3Cchar%3E%3E()%3B%0A%20%20%20%20println!(%22%7B%7D%22%2C%20chars.len())%3B%20%2F%2F%20ar%20trebui%20sa%20fie%207%0A%20%20%20%20%2F%2F%20cum%20un%20char%20ocupa%204%20biti%2C%20putem%20converti%20la%20u32%0A%20%20%20%20println!(%22%7B%7D%22%2C%20chars%5B3%5D%20as%20u32)%3B%0A%7D%0A
+- title: Șiruri de caractere
+  content_markdown: >
+    Un șir de caractere (în engleză *string*) este o structură care deține o secvență de octeți utf-8 în memoria heap.
+
+
+    Deoarece memoria sa se află în heap, aceasta poate fi extinsă, modificată etc. în moduri în care șirurile
+    de caractere literale nu pot fi modificate.
+
+
+    Metode comune:
+
+
+    * `push_str` pentru a adăuga mai mulți octeți utf-8 la sfârșitul unui șir de caractere.
+
+    * `replace` pentru a înlocui secvențe de octeți utf-8 cu altele.
+
+    * `to_lowercase`/`to_uppercase` pentru a schimba între majuscule și minuscule.
+
+    * `trim` pentru eliminarea spatiilor.
+
+
+    Atunci când un șir de caractere este eliminat (drop), este eliminată și memoria heap a acestuia.
+
+
+    Șirile de caractere au un operator `+` care extinde șirul cu un `&str` și  se returnează pe sine,
```

Comment:
Detail this.

Reply:
i tried to add some details in the last commit. im not sure if it is enough

Reply:
i dont really understand the "ergonomics" part on the following line, but it was in the original english text so i kept it